inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,145 @@ See the OMA book for examples.
-
-# Plot the reduced dimensions of pcoa_bray with clusters as colors
-plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
-
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+#| label: heatmap-visualization-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
+library(ggplot2)
+library(cluster)
+
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+se <- SummarizedExperiment(assays = list(counts = assay(GlobalPatterns)),
+                           colData = colData(GlobalPatterns),
+                           rowData = rowData(GlobalPatterns))
+
+# Perform PCoA on the abundance data
+distance_matrix <- vegdist(t(assay(se)), method = "bray")
+pcoa_results <- cmdscale(distance_matrix, k = 2)
+pcoa_data <- data.frame(pcoa_results)
+colnames(pcoa_data) <- c("PCoA1", "PCoA2")
+
+# Perform k-means clustering (k = 3)
+set.seed(123)
+kmeans_result <- kmeans(pcoa_data, centers = 3)
+pcoa_data$cluster <- as.factor(kmeans_result$cluster)
+
+# Visualize PCoA with clusters
+ggplot(pcoa_data, aes(x = PCoA1, y = PCoA2, color = cluster)) +
+  geom_point(size = 3) +
+  labs(title = "PCoA with k-means Clustering", x = "PCoA Axis 1", y = "PCoA Axis 2") +
+  theme_minimal() +
+  scale_color_manual(values = c("red", "blue", "green"))
 ```
-Where : 
-'reducedDim' is a function that extracts the reduced dimensionality representation of your data (e.g., from a PCA or 
-PCoA object).
-'plotReducedDim' is a function that can plot these reduced dimensions and has a 'colour_by argument' that accepts a vector of cluster assignments.
 
-#### 2. Heatmap with Annotations:
+### 2: Cluster Analysis with Heatmap Visualization
 
-* Annotate the heatmap with additional metadata (e.g., subject groups or sample types).
+- Hierarchical clustering of samples based on abundance data.
+- Visualizes the clustered data using a heatmap, with cluster information added to metadata.
 
 ```{r}
+#| label: heatmap-visualization-ex1
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
 # Load necessary libraries
+library(mia)
 library(SummarizedExperiment)
 library(pheatmap)
+library(cluster)
+
+# Generate a sample SummarizedExperiment object 
+data(GlobalPatterns)
+se <- SummarizedExperiment(assays = list(counts = assay(GlobalPatterns)),
+                           colData = colData(GlobalPatterns),
+                           rowData = rowData(GlobalPatterns))
+
+# Extract the abundance data matrix
+abundance_data <- assay(se, "counts")
+
+# Perform hierarchical clustering on the samples
+distance_matrix <- dist(t(abundance_data), method = "euclidean")
```

Comment:
Usually this is not done for counts + euclidean. Common choice is CLR + euclidean or relative abundance + Bray-Curtis